cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterUtilsTest.java

```diff
@@ -136,4 +141,39 @@ public void testParseExecutionProgressCheckIntervalMsWithValue() {
     EasyMock.verify(mockRequest);
     Assert.assertEquals(Long.valueOf(EXECUTION_PROGRESS_CHECK_INTERVAL_STRING), executionProgressCheckIntervalMs);
   }
+
+  @Test
+  public void testGetEndpoint() {
+    verifyGetEndpoint("/a/*");
+    verifyGetEndpoint(WebServerConfig.DEFAULT_WEBSERVER_API_URLPREFIX);
+  }
+
+  /**
+   * Verifies that {@link ParameterUtils#endPoint} can parse out all endpoints
+   *
+   * @param apiUrlPrefixParam API URL prefix parameter passed to the {@link ParameterUtils#endPoint}
+   */
+  private void verifyGetEndpoint(String apiUrlPrefixParam) {
+    HttpServletRequest mockRequest = EasyMock.mock(HttpServletRequest.class);
+    String apiUrlPrefix = apiUrlPrefixParam.substring(0, apiUrlPrefixParam.length() - 1);
+    for (CruiseControlEndPoint getEndPoint : CruiseControlEndPoint.getEndpoints()) {
+      String mockRequestUri = apiUrlPrefix + getEndPoint;
+      EasyMock.expect(mockRequest.getMethod()).andReturn(GET_METHOD).times(1);
+      EasyMock.expect(mockRequest.getRequestURI()).andReturn(mockRequestUri).times(1);
+      EasyMock.replay(mockRequest);
+      CruiseControlEndPoint endPoint = ParameterUtils.endPoint(mockRequest, apiUrlPrefixParam);
+      Assert.assertEquals(getEndPoint, endPoint);
+      EasyMock.reset(mockRequest);
+    }
+
+    for (CruiseControlEndPoint postEndPoint : CruiseControlEndPoint.postEndpoints()) {
+      String mockRequestUri = apiUrlPrefix + postEndPoint;
+      EasyMock.expect(mockRequest.getMethod()).andReturn(POST_METHOD).times(1);
+      EasyMock.expect(mockRequest.getRequestURI()).andReturn(mockRequestUri).times(1);
+      EasyMock.replay(mockRequest);
+      CruiseControlEndPoint endPoint = ParameterUtils.endPoint(mockRequest, apiUrlPrefixParam);
+      Assert.assertEquals(postEndPoint, endPoint);
+      EasyMock.reset(mockRequest);
```

Comment:
Does `reset` call also `verify` the mock? -- i.e. if not do we need `EasyMock.verify()`? (similar use above)

Reply:
I will call `verify` before `reset`